lms/djangoapps/learner_dashboard/serializers.py

```diff
@@ -14,11 +14,126 @@ class PlatformSettingsSerializer(serializers.Serializer):
     courseSearchUrl = serializers.URLField()
 
 
+class CourseProviderSerializer(serializers.Serializer):
+    """Info about a course provider (institution/business)"""
+
+    name = serializers.CharField()
+    website = serializers.URLField()
+    email = serializers.EmailField()
+
+
+class CourseSerializer(serializers.Serializer):
+    """Serializer for course header info"""
+
+    bannerImgSrc = serializers.URLField()
+    courseName = serializers.CharField()
+
+
+class CourseRunSerializer(serializers.Serializer):
+    """Serializer for course run info"""
+
+    isPending = serializers.BooleanField()
+    isStarted = serializers.BooleanField()
+    isFinished = serializers.BooleanField()
+    isArchived = serializers.BooleanField()
+    courseNumber = serializers.CharField()
+    accessExpirationDate = serializers.DateTimeField()
+    minPassingGrade = serializers.DecimalField(max_digits=5, decimal_places=2)
+    endDate = serializers.DateTimeField()
+    homeUrl = serializers.URLField()
+    marketingUrl = serializers.URLField()
+    progressUrl = serializers.URLField()
+    unenrollUrl = serializers.URLField()
+    upgradeUrl = serializers.URLField()
+
+
 class EnrollmentSerializer(serializers.Serializer):
-    """Serializer for an enrollment"""
+    """Info about this particular enrollment"""
+
+    isAudit = serializers.BooleanField()
```

Comment:
Contract question, i know, but do you know why we're doing this rather than having a `mode`?

Reply:
@muselesscreator ?